### ISSUE TITLE
refetch gets undefined when state changes

### ISSUE THREAD

**Issue by kickbk** [apollographql/apollo-client#8670]
Using Expo 42 and apollo-client 3.4.8, I'm getting `TypeError: undefined is not an object (evaluating 'refetch().then')` when the `user` state updates and runs the `useEffect` below.

```js
import {
  getCompletedOrders,
  useLazyQuery,
  useMutation,
} from "dineden-graphql";

const Screen = () => {
  const user = useStore((state) => state.user);
  const [GQL_getCompletedOrders, { data, refetch, loading }] = useLazyQuery(getCompletedOrders);

  useEffect(() => {
    if (user) {
      GQL_getCompletedOrders({
        variables: {
          uid: user?.id,
        },
      });
    }
  }, [GQL_getCompletedOrders, user]);

    useEffect(() => {
    if (refetch && data && user?.orders?.new_completed_order) {
      refetch().then((response) => { ///// <<< BREAKS HERE
        ...
      });
    }
  }, [data, refetch, user]);

  ...
  ```
  
 Is the state update setting `refetch` to undefined? what is actually going on here? I saw a bunch of related posts here about this, but I could not find a definitive answer. Please advise.

**Comment by brainkim** [comment]
This should be fixed in the 3.5 beta. `npm install @apollo/client@beta`.

**Comment by hwillson** [comment]
This should be resolved in @apollo/client@latest - let us know if not.
